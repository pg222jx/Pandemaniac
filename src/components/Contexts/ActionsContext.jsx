import React,{ useState, createContext } from 'react';

export const ActionsContext = createContext()

export const ActionsProvider = props => {

    const [amount, setAmount] = useState(2)

    return (
        <ActionsContext.Provider value={[amount, setAmount]}>
            {props.children}
        </ActionsContext.Provider>
    )
}

