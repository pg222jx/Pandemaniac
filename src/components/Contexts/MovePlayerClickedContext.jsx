import React,{ useState, createContext } from 'react';

export const MovePlayerClickedContext = createContext()

export const MovePlayerClickedProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <MovePlayerClickedContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </MovePlayerClickedContext.Provider>
    );
}

