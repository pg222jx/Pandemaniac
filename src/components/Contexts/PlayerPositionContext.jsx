import React,{ useState, createContext } from 'react';

export const PlayerPositionContext = createContext()

export const PlayerPositionProvider = props => {

    const [position, setPosition] = useState({bottom: undefined, left: undefined})

    return (
        <PlayerPositionContext.Provider value={[position, setPosition]}>
            {props.children}
        </PlayerPositionContext.Provider>
    )
}

