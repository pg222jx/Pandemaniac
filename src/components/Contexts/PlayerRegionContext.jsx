import React,{ useState, createContext } from 'react';

export const PlayerRegionContext = createContext()

export const PlayerRegionProvider = props => {

    const [region, setRegion] = useState('')

    return (
        <PlayerRegionContext.Provider value={[region, setRegion]}>
            {props.children}
        </PlayerRegionContext.Provider>
    )
}

