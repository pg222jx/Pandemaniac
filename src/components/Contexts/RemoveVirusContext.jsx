import React,{ useState, createContext } from 'react';

export const RemoveVirusContext = createContext()

export const RemoveVirusProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <RemoveVirusContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </RemoveVirusContext.Provider>
    );
}

