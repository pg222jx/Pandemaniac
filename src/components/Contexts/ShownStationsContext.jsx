import React, { useState, createContext } from 'react'

export const ShownStationsContext = createContext()

export const ShownStationsProvider = props => {

    const [shownStations, setShownStations] = useState([])

    return (
        <ShownStationsContext.Provider value={[shownStations, setShownStations]}>
            {props.children}
        </ShownStationsContext.Provider>
    )
}

