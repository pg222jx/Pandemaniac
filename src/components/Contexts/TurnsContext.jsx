import React,{ useState, createContext } from 'react';

export const TurnsContext = createContext()

export const TurnsProvider = props => {

    const [amount, setAmount] = useState(0)

    return (
        <TurnsContext.Provider value={[amount, setAmount]}>
            {props.children}
        </TurnsContext.Provider>
    )
}

