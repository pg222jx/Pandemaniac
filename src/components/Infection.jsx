/**
 * Representing the infection phase.
 *
 * @author Pernilla GÃ¶th
 * @version 1.0.0
 */


import React, { useContext } from 'react'
import { ActionsContext } from './Contexts/ActionsContext'
import { ShownStationsContext } from './Contexts/ShownStationsContext'
import { TurnsContext } from './Contexts/TurnsContext'


const Infection = (props) => {
const [actions, setActions] = useContext(ActionsContext) 
const [shownStations, setShownStations] = useContext(ShownStationsContext) // eslint-disable-line no-unused-vars
const [turnsAmount, setTurnsAmount] = useContext(TurnsContext)

const chancePerStation = 5

/**
 * Checks amount of stations rendered and calculates the risk of a virus spawning.
 */
const  infect = async () => {
    const amountOfStations = shownStations.length
    const randomNr = [Math.floor(Math.random() * 100)]
    let chance;

    if (actions < 2) {
        if (await amountOfStations === 6) {
            props.onWin()
        } else {
            if (await amountOfStations === 5) {
                chance = 40
            } else {
                chance = amountOfStations * chancePerStation
            }
    
            if (chance < randomNr) {
                props.spawnRandomVirus()            
            } else {
                props.information.textContent = 'No continent was infected this turn!'
            }
            
            setActions(2)
            setTurnsAmount(turnsAmount + 1)
        }
    }
}

    return ( 
        <button className="Turn" id="done" onClick={() => {infect()}}>Done with turn!</button>
     )
}
 
export default Infection
