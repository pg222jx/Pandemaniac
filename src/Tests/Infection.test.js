import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import {  ActionsProvider } from '../components/Contexts/ActionsContext'
import {  ShownStationsProvider } from '../components/Contexts/ShownStationsContext'
import {  TurnsProvider } from '../components/Contexts/TurnsContext'
import Infection from '../components/Infection'

  const wrapperShallow = shallow(
    <ActionsProvider>
        <ShownStationsProvider>
            <TurnsProvider>
                <Infection />
            </TurnsProvider>
        </ShownStationsProvider>
    </ActionsProvider>
)

  describe('Testing the Infection component', () => {
    it('should match the snapshot', () => {
        expect(wrapperShallow.html()).toMatchSnapshot()
      })

      it('should render without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
        <ActionsProvider>
            <ShownStationsProvider>
                <TurnsProvider>
                    <Infection />
                </TurnsProvider>
            </ShownStationsProvider>
        </ActionsProvider>, div
        )
       })
  })