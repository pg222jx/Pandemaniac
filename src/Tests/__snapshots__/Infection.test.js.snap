// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Infection component should match the snapshot 1`] = `"<button class=\\"Turn\\" id=\\"done\\">Done with turn!</button>"`;
